apollo-router/src/plugins/override_url.rs

```diff
@@ -0,0 +1,109 @@
+use apollo_router_core::{register_plugin, Plugin, SubgraphRequest, SubgraphResponse};
+use reqwest::Url;
+use std::collections::HashMap;
+use tower::util::BoxService;
+use tower::{BoxError, ServiceExt};
+
+#[derive(Debug)]
+struct OverrideSubgraphUrl {
+    urls: HashMap<String, Url>,
+}
+
+#[async_trait::async_trait]
+impl Plugin for OverrideSubgraphUrl {
+    type Config = HashMap<String, Url>;
+
+    fn new(configuration: Self::Config) -> Result<Self, BoxError> {
+        tracing::info!("OverrideSubgraphUrl {:#?}!", configuration);
+        Ok(OverrideSubgraphUrl {
+            urls: configuration,
+        })
+    }
+
+    fn subgraph_service(
+        &mut self,
+        subgraph_name: &str,
+        service: BoxService<SubgraphRequest, SubgraphResponse, BoxError>,
+    ) -> BoxService<SubgraphRequest, SubgraphResponse, BoxError> {
+        let mut new_url = self.urls.get(subgraph_name).cloned();
+        service
+            .map_request(move |mut req: SubgraphRequest| {
+                if let Some(new_url) = new_url.take() {
+                    req.http_request
+                        .set_url(new_url)
+                        .expect("url has been checked when we configured the plugin");
+                }
+
+                req
+            })
+            .boxed()
+    }
+}
+
+register_plugin!(
+    "com.apollographql",
```

Comment:
Let's leave the group blank for our plugins. We're core.

Reply:
Ok that's was also my feeling and then I looked up for the `reporting` plugin. I think it could be worth to do the same for `reporting` plugin.

Reply:
Agreed

Reply:
As @garypen rightfully pointed out [in this comment](https://github.com/apollographql/router/pull/532#discussion_r819401449) we definitely want an ADR about the naming convention

Reply:
opened https://github.com/apollographql/router/issues/581